mitmproxy/io/proto/http.proto

```diff
@@ -0,0 +1,93 @@
+syntax='proto2';
+
+message HTTPFlow {
+    optional HTTPRequest request = 1;
+    optional HTTPResponse response = 2;
+    optional HTTPError error = 3;
+    optional ClientConnection client_conn = 4;
+    optional ServerConnection server_conn = 5;
+    optional bool intercepted = 6;
+    optional bool marked = 7;
+    optional string mode = 8;
+    optional string id = 9;
+}
+
+message HTTPRequest {
+    optional string first_line_format = 1;
+    optional string method = 2;
+    optional string scheme = 3;
+    optional string host = 4;
+    optional int32 port = 5;
+    optional string path = 6;
+    optional string http_version = 7;
+    repeated HTTPHeader headers = 8;
+    optional bytes content = 9;
```

Comment:
The content here and in response will often dwarf the rest of the items in this message. When we iterate over flows to show indexes or do searches, we will often not need to read the content, just the surrounding data. I think the content objects should go in a separate large object table and retrieved as needed.

Reply:
For some reason I decoupled the message definition  from the discussions about view and sessions, only focusing on reproducing state object's behavior. Obviously, it wouldn't make sense to parse and save every time HTTP body. I'll move that out in its own message.